pkgs/by-name/up/upscaler/package.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  fetchFromGitLab,
+  meson,
+  ninja,
+  gitUpdater,
+  desktop-file-utils,
+  appstream,
+  blueprint-compiler,
+  python3Packages,
+  pkg-config,
+  libadwaita,
+  wrapGAppsHook4,
+  upscayl-ncnn,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "upscaler";
+  version = "1.4.0";
+
+  pyproject = false; # meson instead of pyproject
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Upscaler";
+    rev = version;
+    hash = "sha256-Dy8tykIbK5o0XulurG+TxORZZSxfRe5Kjh6aGpsh+0Y=";
+  };
+
+  passthru.updateScript = gitUpdater { };
+
+  postPatch = ''
+    substituteInPlace upscaler/window.py \
+      --replace-fail '"upscayl-bin",' '"${lib.getExe upscayl-ncnn}",'
+  '';
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    meson
+    ninja
+    desktop-file-utils
+    appstream
+    blueprint-compiler
+    pkg-config
+  ];
+
+  dependencies = with python3Packages; [
+    pygobject3
+    pillow
+    vulkan
+  ];
+
+  buildInputs = [
+    libadwaita
+    upscayl-ncnn
+  ];
+
```

```suggestion

  mesonFlags = [
    (lib.mesonBool "network_tests" false)
  ];
  
  # NOTE: `postCheck` is intentionally not used here, as the entire checkPhase
  # is skipped by `buildPythonApplication`
  # https://github.com/NixOS/nixpkgs/blob/9d4343b7b27a3e6f08fc22ead568233ff24bbbde/pkgs/development/interpreters/python/mk-python-derivation.nix#L296
  postInstallCheck = ''
    mesonCheckPhase
  '';
```
Let's make sure we're running any meson tests